src/workflows-app/utils/submission-utils.js

```diff
@@ -181,6 +182,12 @@ const validateRecordLookup = (inputSource, inputType, recordAttributes) => {
     if (!typeMatch(inputType, _.get(`${inputSource.record_attribute}.datatype`)(recordAttributes))) {
       return { type: 'error', message: 'Provided type does not match expected type' };
     }
+    if (
+      unwrapOptional(inputType).type === 'array' &&
+      !_.startsWith('ARRAY_OF_')(_.get(`${inputSource.record_attribute}.datatype`)(recordAttributes))
```

Comment:
A simlar nesting-level question - what if CBAS needs an Array[Array[Int]] and WDS is only providing a single-layer ARRAY_OF_INT? I think that's morally equivalent to the auto-boxing above, but wouldn't trigger this warning message

Reply:
You're right that it wouldn't trigger this message - though I'm not even sure what the behavior would be in this case... I'm assuming we would want to convert e.g. [1, 2, 3] to [[1, 2, 3]] rather than [[1], [2], [3]] but it does get a bit more fuzzy

Reply:
Do we also want to do nested conversions on multiple levels, e.g. Array[Array[Int]] from WDS NUMBER? 1 => [[1]]

Reply:
Yeah I'm happy to leave these all as unreported edge cases for now. For double boxing... I don't know. But I suspect not, because we would probably need to program it as an explicit case (we tend to avoid **_searching_** for indirect coercions (eg x -> y -> z => x -> z) automatically. Partly because the search space could be huge but also because so many things _could_ go indirectly via String (Int -> String -> File) that static type checking would lose its value and everything would become a not super helpful runtime coercion error.)